src/server.js

```diff
@@ -202,6 +209,53 @@ app.get(
   }
 );
 
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_EVENT_ID}/restaurants`,
+  getEvent,
+  async (request, response) => {
+    const { event } = request;
+    const users = event.users || {};
+    const userData = Object.values(users);
+
+    if (userData.length > 0) {
+      // Currently accessing the latitude and longitude of the first user (MVP).
+      // TODO (Chisom): Test average geolocation with multiple user locations.
+      const lat = userData[0].lat.toString();
+      const long = userData[0].long.toString();
+      const radiusMeters = "50000";
+      const type = "restaurant";
+      const minprice = "0";
+      const maxprice = "4";
+
+      const placesApiResponse = await (
```

Comment:
Fetch docs:  https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API, would recommend reading.

There are 3 levels of errors that can/should be handled:
1. Network failure -- this is unlikely, but maybe could happen if we mess up the endpoint URL. In this scenario, `fetch` will throw an exception (when using `await`), so we need `try`/`catch`
2. 404/500 error. Once again this happens when we mess-up the api URL. We can't assume the response is parse-able into JSON using the `json` function. So we need to check if the response is `ok` before parsing it into JSON.

> The Promise returned from fetch() won’t reject on HTTP error status even if the response is an HTTP 404 or 500. Instead, it will resolve normally (with ok status set to false), and it will only reject on network failure or if anything prevented the request from completing.

3. Parameter error. Maybe the response is valid JSON, but the `status` is still invalid. We need to check for this condition too.

Reply:
To clarify, `fetch` can throw (but it only throws when the request failed to complete due to a timeout / bad network).

And, even if fetch does not throw, we cannot assume the response is valid JSON (it could be something like "Cannot GET /api"). So we need to check `response.ok`. If `ok` is `true`, then we can assume the response is valid JSON. `ok` will generally only be true when the status code is 200. (I don't know the exact details, but I believe this is the case).

And, even if the response is valid JSON then we can't necessarily assume it completed successfully (well if the response is 200, then yeah we probably can), but just to be safe/paranoid, we should check the actual status inside the JSON itself.

Reply:
Given all of this, the right thing to do is probably:
- Wrap the `fetch` and the `json()` in a `try`/`catch`. If that fails, we know we didn't get a response that was valid JSON.
- **Also** check the response status code inside the resulting JSON. If it's not `"OK"`, something's wrong.